clouddriver-appengine/src/main/groovy/com/netflix/spinnaker/clouddriver/appengine/gcsClient/AppengineGcsRepositoryClient.groovy

```diff
@@ -46,13 +47,29 @@ class AppengineGcsRepositoryClient implements AppengineRepositoryClient {
     }
 
     def dest = targetDirectory + File.separator + applicationDirectoryRoot
-    def fullPath = repositoryUrl.substring(gsPrefix.length()) + '/' + applicationDirectoryRoot
-    def slash = fullPath.indexOf("/");
-    def bucketName = fullPath.substring(0, slash);
-    def bucketPath = fullPath.substring(slash + 1);
+
+    def fullPath = repositoryUrl.substring(gsPrefix.length())
+    if (applicationDirectoryRoot) {
+      fullPath + '/' + applicationDirectoryRoot
+    }
+    def slash = fullPath.indexOf("/")
+    def bucketName = fullPath.substring(0, slash)
+    def bucketPath = fullPath.substring(slash + 1)
 
     if (fullPath.endsWith(".tar")) {
-      StorageUtils.untarStreamToPath(storage.openObjectStream(bucketName, bucketPath), dest)
+      InputStream tas = storage.openObjectStream(bucketName, bucketPath)
+
+      // NOTE: We write the tar file out to an intermediate temp file because the tar input stream
```

Comment:
:|

Reply:
You'd think someone would have solved this before...